engine-tests/src/test/java/org/terasology/documentation/ModuleMarkupScraper.java

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.documentation;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import org.codehaus.plexus.util.StringUtils;
+import org.terasology.HeadlessEnvironment;
+import org.terasology.context.Context;
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.TerasologyConstants;
+import org.terasology.engine.module.ExtraDataModuleExtension;
+import org.terasology.engine.module.ModuleManager;
+import org.terasology.engine.module.RemoteModuleExtension;
+import org.terasology.engine.module.StandardModuleExtension;
+import org.terasology.entitySystem.event.Event;
+import org.terasology.i18n.I18nMap;
+import org.terasology.module.*;
+import org.terasology.naming.Name;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import static org.terasology.engine.module.StandardModuleExtension.IS_ASSET;
+import static org.terasology.engine.module.StandardModuleExtension.IS_WORLD;
+
+public final class ModuleMarkupScraper {
+
+    private static final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private static final String DEFAULT_GITHUB_MODULE_URL = "https://github.com/Terasology/";
+    private static final List INTERNAL_MODULES = Arrays.asList("Core", "engine", "CoreSampleGameplay", "BuilderSampleGameplay", "BiomesAPI");
```

Comment:
Hi, Thanks for the input!

I did see the list in the scrapper. However, I was fearful that it was not up to date. More accuracy, given I'm new to your project I just didn't know what I was doing (lol). So I thought a safer approach was to generate the modules to help maintainability going forward?

Additionally, given earlier feedback I realized `HEnv ` was overkill. I did a bit more digging. All I really needed was the RemoteRegistry.  So I created an extra factory method ModuleManagerFactory.create("meta.terasology.org") so that the default modelManager would initialize properly.

The result is a list of modules identical to those you would expect to see on the "Advanced Game Setup" and "Module Details" in game screens.

Again, maybe that dynamic approach is unnecessary. I can go back to the INTERNAL_MODULES approach if that is whats preferable (no issue either way).

Reply:
Oh, I thought this was used to filter out the internal modules.

I'm also not sure whether the dynamic approach is what we should do, but it's better than nothing. Overall, I would like to keep the PRs small and restricted to on change set - there should be nothing in this PR that does not belong to the dynamic module fetching approach, for instance.

Reply:
The discussion in Discord agrees with your concerns with this dynamic approach. Until there is agreement, I think perhaps I should refactor but avoid adding features unless instructed otherwise. I can look to the issue list to find another task?

Reply:
> Also, what's the easiest way to run this? Do we have a simple Gradle task or command? I just executed the main method via IntelliJ, but that's not an option for the CI, and it's also easy to miss/forget...

I've been running it as a command line. I'd think a Gradle task or via command line?